src/Processor/Braintree/Subscription/Update/ChangeBillingCycleStrategy.php

```diff
@@ -64,12 +68,36 @@ private function createReplacementSubscription(Subscription $original, Subscript
         $addOns = $newSubscription->getAddOnsImpl();
 
         return $newSubscription
-            ->merge($canceled)
+            ->merge($original)
             ->setCustomer($original->getCustomer())
             ->setAddOns($addOns)
             ->setDiscounts($discounts);
     }
 
+    private function getNewPurchaseModifier(Subscription $sub): ?Modifier
+    {
+        $balance = $sub->getBalance();
+        $fuzzy = $sub->getRemainingValue()->getAmount();
+
+        if (null === $balance || 0.0 === $balance->getAmount()) {
+            $fuzzy = $fuzzy;
+        } elseif ($balance->getAmount() < 0.0) {
+            $fuzzy += abs($balance->getAmount());
+        } elseif ($balance->getAmount() > 0.0) {
+            $fuzzy -= abs($balance->getAmount());
+        }
+
+        if (0.0 === $fuzzy) {
+            return null;
+        }
+
+        if ($fuzzy >= 0.0) {
+            return new Discount('balance', 1, new Price($fuzzy));
+        }
+
+        return new AddOn('overdue', 1, new Price(abs($fuzzy)));
```

Comment:
@johncorrelli we don't have a test case for this in the suite, but we never did. Can you spot check this for correctness?